Sources/SwiftDocC/Checker/Checkers/TopicWithoutSubheading.swift

```diff
@@ -0,0 +1,72 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import Markdown
+
+/**
+ A `Topic` should have at least one subheading.
+ */
+public struct TopicWithoutSubheading: Checker {
+    public var problems = [Problem]()
+
+    private var sourceFile: URL?
+
+    /// Creates a new checker that detects Topics without subheadings.
+    ///
+    /// - Parameter sourceFile: The URL to the documentation file that the checker checks.
+    public init(sourceFile: URL?) {
+        self.sourceFile = sourceFile
+    }
+
+    public mutating func visitDocument(_ document: Document) -> () {
+        let headings = getHeadings(document: document)
+        for (index, element) in headings.enumerated() {
+            guard element.title == "Topics" else {
+                continue
+            }
+
+            if !hasSubheading(heading: element, subnodes: headings.dropFirst(index + 1)) {
+                warnHeading(element)
+            }
+        }
+    }
+
+    private func getHeadings(document: Document) -> [Heading] {
+        return (0..<document.childCount).compactMap { index -> Heading? in
```

Comment:
This can be simplified by mapping over `document.children` instead of the indices:
```swift
return document.children.compactMap { $0 as? Heading }
```
With that implementation I don’t feel that the private function, which is only called once, improves the overall readability.